plugins/catalog-backend/src/database/search.ts

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { DescriptorEnvelope } from '../ingestion';
+import { DbEntitiesSearchRow } from './types';
+
+// Search entries that start with these prefixes, also get a shorthand without
+// that prefix
+const SHORTHAND_KEY_PREFIXES = [
+  'metadata.',
+  'metadata.labels.',
+  'metadata.annotations.',
```

Comment:
Not crazy about allowing lookup based on annotations

Reply:
I think it'll be used. We can strip it out if we want but not sure if it'll actually hurt.